channeldb/invoices.go

```diff
@@ -723,28 +723,16 @@ func fetchInvoiceNumByRef(invoiceIndex, payAddrIndex kvdb.RBucket,
 	}
 }
 
-// InvoiceWithPaymentHash is used to store an invoice and its corresponding
-// payment hash. This struct is only used to store results of
-// ChannelDB.FetchAllInvoicesWithPaymentHash() call.
-type InvoiceWithPaymentHash struct {
-	// Invoice holds the invoice as selected from the invoices bucket.
-	Invoice Invoice
-
-	// PaymentHash is the payment hash for the Invoice.
-	PaymentHash lntypes.Hash
-}
-
-// FetchAllInvoicesWithPaymentHash returns all invoices and their payment hashes
-// currently stored within the database. If the pendingOnly param is true, then
-// only open or accepted invoices and their payment hashes will be returned,
-// skipping all invoices that are fully settled or canceled. Note that the
-// returned array is not ordered by add index.
-func (d *DB) FetchAllInvoicesWithPaymentHash(pendingOnly bool) (
-	[]InvoiceWithPaymentHash, error) {
-
-	var result []InvoiceWithPaymentHash
+// ScanInvoices scans trough all invoices and calls the passed scanFunc for
+// for each invoice with its respective payment hash. The scanFunc also receives
+// a scanCount integer parameter in case the kvdb.View function is retried.
+func (d *DB) ScanInvoices(
+	scanFunc func(int, lntypes.Hash, *Invoice) error) error {
 
+	scanCount := 0
 	err := kvdb.View(d, func(tx kvdb.RTx) error {
+		scanCount++
```

Comment:
👍 

So then what's the intended behavior of the caller if this is ever greater than one during closure execution?

Reply:
The caller should clear their partial results as this indicates that the scan is being retried.